app/views/create_project.py

```diff
@@ -0,0 +1,83 @@
+from flask import session, redirect, url_for, request, render_template, flash
+from app import app, forms
+from app.util import is_logged_in, error_message
+import pymesync
+import re
+
+
+@app.route('/projects/create/', methods=['GET', 'POST'])
+def create_project():
+    if not is_logged_in():
+        if request.method == 'GET':
+            return redirect(url_for('login', next=request.url_rule))
+        elif request.method == 'POST':
+            return "Not logged in.", 401
+
+    form = forms.CreateProjectForm()
+
+    ts = pymesync.TimeSync(baseurl=app.config['TIMESYNC_URL'],
+                           test=app.config['TESTING'],
+                           token=session['token'])
+
+    users = ts.get_users()
+    error_message(users)
+
+    usernames = []
+    for user in users:
+        usernames.append((user['username'], user['display_name']))
+
+    # Fill the multi-select boxes with usernames
+    # In order to be able to select users for each permission, you must have a
+    # list of users
+    form.members.choices = usernames
+    form.managers.choices = usernames
+    form.spectators.choices = usernames
+
+    if form.validate_on_submit():
+        project = {}
+        permissions = {}
+
+        for field in form:
+            if field.data and field.name is not 'csrf_token':
+                if field.name == 'slugs':
+                    project['slugs'] = re.split('\s?,\s?', form.slugs.data)
+                elif field.name == 'members':
```

Comment:
The code to turn the permissions lists into a permissions dictionary could probably be made less nested and complicated. Maybe it could look something like this:

```python
users = set(request.form['members'] + request.form['managers'] + request.form['spectators'])
for user in users:
    user_permissions = {
        "member": user in request.form['members'],
        "manager": user in request.form['managers'],
        "spectator": user in request.form['spectators']
    }
    permissions[user] = user_permissions
```

Since it appears both here and in edit_project.py maybe it should also be factored out into a function inside util.py